### PR TITLE
Added check for existing git install location

### PR DESCRIPTION
**Pull request by areyling**
Fix for ferventcoder/chocolateyautomaticpackages#79

### CODE REVIEW

File: git.install/tools/chocolateyInstall.ps1 [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+
+  $key = 'HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Git_is1'
+  #adjust for 64-bit Windows (explicitly read from 32-bit reg)
+  if (-not (Test-Path $key)) { $key = 'HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall\Git_is1' }
+  if (Test-Path $key) {
+    $path = (Get-ItemProperty -Path $key -Name 'InstallLocation').InstallLocation
+    $silentArgs += ' /DIR=' + "`"$path`""
+  }
+  
+  Install-ChocolateyPackage 'git.install' 'exe' $silentArgs '{{DownloadUrl}}'
```
Comment (ferventcoder) [inline git.install/tools/chocolateyInstall.ps1]: $silentargs should still be in double quotes.

**Comment by areyling** [comment]
I'm assuming you mean:

```
Install-ChocolateyPackage 'git.install' 'exe' "$silentArgs" '{{DownloadUrl}}'
```

It works for me locally [the way it is now](https://github.com/areyling/chocolateyautomaticpackages/blob/ea01355bcdf248351e359f387b4ac25d6b8be4b9/git.install/tools/chocolateyInstall.ps1#L12), but also as shown above.  Is this to ensure compatibility?

**Comment by ferventcoder** [comment]
Yes

On Saturday, June 7, 2014, Andrew Reyling notifications@github.com wrote:

> I'm assuming you mean:
> 
> Install-ChocolateyPackage 'git.install' 'exe' "$silentArgs" '{{DownloadUrl}}'
> 
> It works for me locally the way it is now
> https://github.com/areyling/chocolateyautomaticpackages/blob/ea01355bcdf248351e359f387b4ac25d6b8be4b9/git.install/tools/chocolateyInstall.ps1#L12,
> but also as shown above. Is this to ensure compatibility?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ferventcoder/chocolateyautomaticpackages/pull/83#issuecomment-45416307
> .

## 

Rob
"Be passionate in all you do"

http://devlicio.us/blogs/rob_reynolds
http://ferventcoder.com
http://twitter.com/ferventcoder

**Comment by TomOne** [comment]
Good that you’re talking about the double quotes around variables. I never understood why they are necessary around a variable name and I never had problems when omitting them. Can you explain that more in detail?

**Comment by ferventcoder** [comment]
I still do a bunch with Posh v2. I've had mixed results when omitting the double quotes so as a matter of habit I just add them around variables that I'm passing. Keep in mind it probably doesn't matter in most situations where you are passing a variable to a function, but sometimes I'm passing it to something outside of powershell (and then it usually matters). I've also run into a really fun time with quoting to get powershell to run a separate powershell window with a command.

File: git.install/tools/chocolateyInstall.ps1 [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
-  Install-ChocolateyPackage 'git.install' 'exe' '/VERYSILENT /NORESTART /NOCANCEL /SP- /CLOSEAPPLICATIONS /RESTARTAPPLICATIONS /NOICONS  /COMPONENTS="assoc,assoc_sh" /LOG' '{{DownloadUrl}}'
+  $silentArgs = '/VERYSILENT /NORESTART /NOCANCEL /SP- /CLOSEAPPLICATIONS /RESTARTAPPLICATIONS /NOICONS /COMPONENTS="assoc,assoc_sh" /LOG'
+
+  $key = 'HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Git_is1'
+  #adjust for 64-bit Windows (explicitly read from 32-bit reg)
+  if (-not (Test-Path $key)) { $key = 'HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall\Git_is1' }
+  if (Test-Path $key) {
+    $path = (Get-ItemProperty -Path $key -Name 'InstallLocation').InstallLocation
+    $silentArgs += ' /DIR=' + "`"$path`""
+  }
```
Comment (ferventcoder) [inline git.install/tools/chocolateyInstall.ps1:10]: Does update not cover this location? The location check should only be needed when setting the path for the application.

Comment (ferventcoder) [inline git.install/tools/chocolateyInstall.ps1:10]: That is down in additional setup when trying to find the path to set.

**Comment by ferventcoder** [comment]
Closing as this has been moved out of this repo - if you believe this is still something to implement, let's discuss it over at https://github.com/ferventcoder/chocolatey-packages/

**Pull request closed without merging**